dev/integration_tests/flutter_driver_screenshot_test/test_driver/flutter_gold_main_test.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart' hide TypeMatcher, isInstanceOf;
+import 'package:flutter_test/src/buffer_matcher.dart';
+
+Future<void> main() async {
+  FlutterDriver driver;
+  String deviceModel;
+
+  setUpAll(() async {
+    driver = await FlutterDriver.connect();
+    deviceModel = await driver.requestData('device_model');
+  });
+
+  tearDownAll(() => driver.close());
+
+  test('A page with an image screenshot', () async {
+    final SerializableFinder imagePageListTile =
+        find.byValueKey('image_page');
+    await driver.waitFor(imagePageListTile);
+    await driver.tap(imagePageListTile);
+    await driver.waitFor(find.byValueKey('red_square_image'));
+    await driver.waitUntilNoTransientCallbacks();
+
+    // TODO(egarciad): This is currently a no-op on LUCI.
+    // https://github.com/flutter/flutter/issues/49837
+    await expectLater(
+      driver.screenshot(),
+      bufferMatchesGoldenFile('red_square_driver_screenshot__$deviceModel.png'),
```

Comment:
I was wondering as I've been working on https://github.com/flutter/flutter/pull/49815, is there anything in the environment that indicates this is a driver test? Or what device it is running on?

Reply:
Not in the environment, unfortunately. 

`enableFlutterDriverExtension()` has an assert `WidgetsBinding.instance is _DriverBinding`. Maybe we could make `_DriverBinding` public.  

cc @jonahwilliams  do you have any suggestions?

Reply:
Maybe we can just provide a flag then, or include something in the file name. When we get down to the skia client I would like to specify in the parameters that this is a device test and/or the device it is running on, but it may not be necessary.

https://github.com/flutter/flutter/pull/49815/files#r376168906

Reply:
The test is always under a `test_driver` directory. Potentially that could be the heuristic.

Reply:
Perfect. SGTM.